File: nvxapp.client/src/nvx/ClientServer-Service/Infrastructure/Account/Models/user-dealer-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";
import { RolesModel } from "./user-roles-model";


export class UserDealerListModel {
  constructor(
    public descrizione: string | null = "",
    public idAspNetUsers: string = "",
    public idUserDealer: number = 0,
    public mainUser: boolean = false,
    public roleId: string = "",
  ) { }
}
export class UserDealerListInModel {
  
}
export class UserDealerListOutModel extends ModelResult {

  public userDealerList: UserDealerListModel[] ;

}



export class UserDealerEditModel {
  constructor(
    public descrizione: string | null = "",
    public idUserDealer: number = 0,
    public mainUser: boolean = false,

    public mail: string | null = null,
    public pw: string | null = null,
    public roleId: string = "",
  ) { }
}

export class UserDealerGetInModel {
  public id: number;
}
export class UserDealerGetOutModel extends ModelResult {

  public userDealerEdit: UserDealerEditModel;

}
export class UserDealerPutInModel  {

  public userDealerEdit: UserDealerEditModel;

}
export class UserDealerPutOutModel extends ModelResult {

  public userDealerEdit: UserDealerEditModel;

}

File: nvxapp.server.service/ClientServer-Service/infrastructure/Account/Models/UserDealerModel.cs
using nvxapp.server.data.Entities.Public;
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static nvxapp.server.data.Entities.AspNetUsersDataUtil;

namespace nvxapp.server.service.ClientServer_Service.Infrastructure.Account.Models
{
    public class UserDealerModel
    {
        public string? IdAspNetUsers { get; set; }
        public int IdUserDealer { get; set; }
        public string? Descrizione { get; set; }
        public Boolean MainUser { get; set; }
        public string? RoleId { get; set; }
    }
    public class UserDealerListInModel
    {
    }
    public class UserDealerListOutModel : ModelResult
    {
        public List<UserDealerModel> UserDealerList { get; set; }

        public UserDealerListOutModel()
        {
            UserDealerList = new List<UserDealerModel>();
        }
    }



    public class UserDealerEditModel
    {
        public int IdUserDealer { get; set; }
        public string? Descrizione { get; set; } = string.Empty;
        public Boolean MainUser { get; set; }
        public string? Mail { get; set; } = string.Empty;
        public string? Pw { get; set; } = string.Empty;
        public string? RoleId { get; set; }
    }
    public class UserDealerGetInModel
    {
        public int Id { get; set; } = 0;
    }
    public class UserDealerGetOutModel : ModelResult
    {
        public UserDealerEditModel UserDealerEdit { get; set; } = new UserDealerEditModel();
    }
    public class UserDealerPutInModel : ModelResult
    {
        public UserDealerEditModel UserDealerEdit { get; set; } = new UserDealerEditModel();
    }
    public class UserDealerPutOutModel : ModelResult
    {
        public UserDealerEditModel UserDealerEdit { get; set; } = new UserDealerEditModel();
    }

}
